### ISSUE TITLE
Update bottomOffset dynamically?

### ISSUE THREAD

**Issue by EugeneLiang** [FaridSafi/react-native-gifted-chat#219]
Hi there!

I was trying to change the `bottomOffset` dynamically but I notice that while the value for `this.state.bottomOffset` changes, `GiftedChat` 's container's `bottomOffset` doesn't change.

Here's a snippet of my code:

```
renderAccessory() {
    return (
      <View style={{backgroundColor:"#000000", height: this.state.bufferHeight, width: width}}>
          <TouchableOpacity
            onPress={() => {
              this.setState({showExtra:!this.state.showExtra})
              if (this.state.showExtra) {
                this.setState({bufferHeight:44})

              }
              else {
                this.setState({bufferHeight:88})

              }

            }}
            >
            <Text style={{color:"#ffffff"}}>FUNNY</Text>
          </TouchableOpacity>

      </View>
    )
}
```

```
render () {
     <GiftedChat
          isAnimated={true}
          messages={this.state.messages}
          onSend={this.onSend}
          loadEarlier={true}
          isLoadingEarlier={true}
          onLoadEarlier={this.onLoadEarlierMessages}
          user={{_id: 1, name:'Thor'}}
          onChangeText={() => this.state.channel ? this.state.channel.typing() : false}
          isLoadingEarlier={this.state.isLoadingEarlierMessages}
          renderBubble={this.renderBubble}
          renderFooter={this.renderFooter}
          bottomOffset={this.state.bufferHeight} // set initially at 44
          renderAccessory={this.renderAccessory}

        />
}
```

What I am trying to do is to increase the height of the `view` component generated by `renderAccessory()`, and as this happen, i need to shift the `GiftedChat` component upwards and hence increasing the `bottomOffset` to 88.

While `console.log` are printing the values correctly, the `borromOffset` does not result in a dynamic change in the height of `GiftedChat`

Am I missing anything?

Thanks!

**Comment by indivisable** [comment]
Same issue for me, cannot seem to set the bottomOffset even when not dynamic (mapped to state) etc.

**Comment by kfiroo** [comment]
@EugeneLiang @indivisable There is some mixup with how `GiftedChat` transfers its props to its state.
As a workaround, you can try to call `setBottomOffset()` on your `GiftedChat` instance

**Comment by indivisable** [comment]
@kfiroo what is best way to reference it as an instance?

I tried using ref but it does not seem to apply the ref="" prop...

```
<GiftedChat
        ref="chatWindow"
...

componentDidMount() {
this.refs.chatWindow.setBottomOffset(100);
...
```

**Comment by kfiroo** [comment]
@indivisable I have to say I'm not completely sure how this works, but looking at the code [here](https://github.com/FaridSafi/react-native-gifted-chat/blob/master/src/GiftedChat.js#L212) it looks like you can simply change the `bottomOffset` prop and when `onKeyboardWillShow` is called it will use that props to calculate the height of the message container.

**Comment by RobAWilkinson** [comment]
@kfiroo did you fix this? I'm experiencing the same problem, here's my render function:

```
   return (
      <GiftedChat
        messages={this.state.messages}
        onSend={this.onSend}
        renderSend={args => <SendButton {...args } />}
        renderBubble={args => <ChatBubble {...args} /> }
        renderInputToolbar={args => <InputToolbar {...args} />}
        minInputToolbarHeight={50}
        renderComposer={args => <ChatComposer {...args} />}
        bottomOffset={0}
        user={{
          _id: 1
        }}
      />
    );
```

**Comment by kfiroo** [comment]
@RobAWilkinson This might help: https://github.com/FaridSafi/react-native-gifted-chat/issues/368

**Comment by RobAWilkinson** [comment]
unfortunately, I need to have a tab bar in my application:
I am overriding a lot of the `renderXXX` props, is there a 'nuclear' option to completely remove that gap?

I am using expo to make dev a bit easier for myself, if that makes a difference. 
<img width="394" alt="screen shot 2017-03-30 at 1 35 50 pm" src="https://cloud.githubusercontent.com/assets/1959628/24525074/d5852746-154d-11e7-8400-e7cb6ef80419.png">

**Comment by RobAWilkinson** [comment]
Anyone who is struggling with this!
Tweak the `minInputToolbarHeight` and `bottomOffset` props that you are giving the `GiftedChat`, start with making sure they are equal, and then increase them!

**Comment by jnrepo** [comment]
Having the same issue here. Methods tried:

1. Adjust height of `View` that wraps gifted container on when the keyboard is shown (doesn't work: gifted container will keep the original height; although in the logs its changing)
2. Tried tweaking `minInputToolbarHeight` and `bottomOffset` (code gets very verbose for a simple feature)

Is there a simpler way to do this?

**Comment by wcastand** [comment]
Hi, wanted to know if bottomOffset can now be used dynamically?

This is the issue I have right now, I need to preview some uploaded photos before sending them and delete the preview once it's sent.

The issue is, when I'm sending the msg, the previews are deleted but the bottomOffset is not updated so I get a gap between the keyboard and my send Button.

I tried to use setBottomOffset() but it's doesn't change anything.
I tried the `minInputToolbarHeight` but it's not helping, what I need would eventually be `maxInputToolbarHeight` but not sure it would solve my issue.

Default :
<img width="384" alt="Screenshot 2019-05-10 at 12 07 35" src="https://user-images.githubusercontent.com/2178244/57523330-9c4bc280-731c-11e9-81f7-832309e0f93f.png">
With preview:
<img width="396" alt="Screenshot 2019-05-10 at 12 07 50" src="https://user-images.githubusercontent.com/2178244/57523349-a241a380-731c-11e9-8439-54f76b310a51.png">
Once the message is sent: 
<img width="394" alt="Screenshot 2019-05-10 at 12 07 55" src="https://user-images.githubusercontent.com/2178244/57523360-ab327500-731c-11e9-8296-74ea23d16dd5.png">
what i would expect: 
<img width="413" alt="Screenshot 2019-05-10 at 12 11 35" src="https://user-images.githubusercontent.com/2178244/57523404-cac99d80-731c-11e9-85fd-583ee899a068.png">

Does anyone have an idea? :)
